dataset/test/dataset_test.cpp

```diff
@@ -158,6 +157,48 @@ TEST(DatasetTest, setData_with_and_without_variances) {
   ASSERT_EQ(d.size(), 2);
 }
 
+void check_array_shared(Dataset &ds, const std::string &name,
+                        const DataArray &array,
+                        const bool shared_coord = true) {
+  EXPECT_EQ(ds[name], array);
+  // Data and meta data are shared
+  EXPECT_TRUE(ds[name].data().is_same(array.data()));
+  EXPECT_EQ(ds[name].coords()[Dim::X].is_same(array.coords()[Dim::X]),
+            shared_coord);
+  EXPECT_TRUE(ds[name].masks()["mask"].is_same(array.masks()["mask"]));
+  EXPECT_TRUE(
+      ds[name].attrs()[Dim("attr")].is_same(array.attrs()[Dim("attr")]));
+  // Metadata *dicts* are not shared
+  ds.coords().erase(Dim::X);
+  EXPECT_NE(ds[name].coords(), array.coords());
+  EXPECT_TRUE(array.coords().contains(Dim::X));
+  ds[name].masks().erase("mask");
+  EXPECT_NE(ds[name].masks(), array.masks());
+  EXPECT_TRUE(array.masks().contains("mask"));
+  ds[name].attrs().erase(Dim("attr"));
+  EXPECT_NE(ds[name].attrs(), array.attrs());
+  EXPECT_TRUE(array.attrs().contains(Dim("attr")));
```

Comment:
Is there a public interface for comparing the VariableConcepts?

Reply:
No there is not. Which aspect are you referring to in this context?

Reply:
It is a bit awkward to test if two objects share memory by modifying one and checking if the other changed. It would be nicer if we could ask them directly if the buffers are the same.

Reply:
There is `Variable::is_same` which was added recently, it is just not used widely yet.